core/src/test/scala/unit/kafka/producer/AsyncProducerTest.scala

```diff
@@ -391,26 +391,27 @@ class AsyncProducerTest {
     // entirely.  The second request will succeed for partition 1 but fail for partition 0.
     // On the third try for partition 0, let it succeed.
     val request1 = TestUtils.produceRequestWithAcks(List(topic1), List(0, 1), messagesToSet(msgs), acks = 1,
-      correlationId = 11, timeout = DefaultAckTimeoutMs, clientId = DefaultClientId)
+      correlationId = 5, timeout = DefaultAckTimeoutMs, clientId = DefaultClientId)
```

Comment:
It may be worth adding a comment somewhere in this test to say that these numbers have changed, possibly due to different number of retries and that it is not worth figuring out a better way do derive these numbers since the test is going away.

Reply:
I don't think the test cares about the correlation ids though. I think we probably would not include that if there was an easy way to exclude that field alone.

Reply:
It just looks odd to see a commit with some arbitrary numbers changed in a test and no comments to explain why.

Reply:
I can add more detail to the commit message, that's where things like that should live.

Reply:
Sounds good. I think the one test failure is unrelated, so LGTM.

Reply:
Added a bit more detail on the reason why both tests had to be updated (I previously only mentioned that they had to be updated).